decidim-initiatives/app/services/decidim/initiatives/signature_handler.rb

```diff
@@ -0,0 +1,222 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Initiatives
+    # This is the base class for signature handlers, all implementations
+    # should inherit from it.
+    # Each SignatureHandler is a form that will be used to check if the
+    # signature is valid or not. When it is valid the initiatives votes
+    # defined by the initiative type will be created for the user.
+    #
+    # Feel free to use validations to assert fields against a remote API,
+    # local database, or whatever.
+    #
+    # It also sets two default attributes, `user` and `initiative`.
+    class SignatureHandler < Form
+      include ValidatableAuthorizations
+
+      mimic :initiatives_vote
+
+      # The user that is trying to sign, it is initialized with the
+      # `current_user` from the controller.
+      attribute :user, Decidim::User
+
+      # The initiative to be signed
+      attribute :initiative, Decidim::Initiative
+
+      validates :initiative, :user, presence: true
+      validate :uniqueness
+      validate :valid_metadata
+      validate :valid_authorized_scopes
+
+      delegate :promote_authorization_validation_errors, :authorization_handler_form_class, to: :workflow_manifest
+      delegate :scope, to: :initiative
+
+      alias signer user
```

Comment:
could be that you are not using `signer` anymore?

Reply:
Removed